dataservice/api/biospecimen/resources.py

```diff
@@ -37,18 +38,27 @@ def get(self, filter_params, after, limit):
             resource:
               Biospecimen
         """
-        # Get study id and remove from model filter params
+        # Get study id, diagnosis_id and remove from model filter params
         study_id = filter_params.pop('study_id', None)
-
+        diagnosis_id = filter_params.pop('diagnosis_id', None)
         # Apply filter params
         q = (Biospecimen.query
              .filter_by(**filter_params))
 
-        # Apply study filter param
         from dataservice.api.participant.models import Participant
-        if study_id:
+        # Apply study_id filter and diagnosis_id filter
+        if diagnosis_id and study_id:
```

Comment:
You should be able to remove this and change the two `elif`s below to `if`s

Reply:
all the three conditions have different joins.  if we remove the `elifs`  two of the conditions will be true at the same time and the query throws multiple joins on the same table.

Reply:
I think they are doing the same joins though?

Reply:
ahh, I see what you are saying if both conditions are satisfied both the conditions queries will be executed.